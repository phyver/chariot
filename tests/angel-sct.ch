data list('x) where
    | Nil : list('x)
    | Cons : 'x -> list('x) -> list('x)

-- shouldn't terminate
val test1 f [] x = []
  | test1 f x [] = []
  | test1 f (x::xs) (y::ys) = test1 f (y::xs) (x::ys)

-- should terminate
val test1 f [] x = []
  | test1 f x [] = []
  | test1 f (x::xs) (y::ys) = test1 f ??? (x::ys)

-- should terminate
val test1 f [] x = []
  | test1 f x [] = []
  | test1 f (x::xs) (y::ys) = test1 f (x::xs) ???

-- shouldn't terminate
val test1 f [] x = []
  | test1 f x [] = []
  | test1 f (x::xs) (y::ys) = test1 ??? (y::xs) (x::ys)

