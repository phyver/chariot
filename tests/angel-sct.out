>> option verbose is set to 1
>> inductive type list was succesfully defined
>> Typing for test1 successful
>> the definition for test1 is complete
!! warning: the definition test1 is NOT provably total (weight_bound: 2, depth_bound: 2)

>> Typing for test2 successful
>> the definition for test2 is complete
>> the definition test2 is provably correct

>> Typing for test3 successful
>> the definition for test3 is complete
>> the definition test3 is provably correct

>> Typing for test4 successful
>> the definition for test4 is complete
!! warning: the definition test4 is NOT provably total (weight_bound: 2, depth_bound: 2)

